blue_hardware:
    ### Joint configuration
    joint_names:
        - left_base_roll_joint
        - left_shoulder_lift_joint
        - left_shoulder_roll_joint
        - left_elbow_lift_joint
        - left_elbow_roll_joint
        - left_wrist_lift_joint
        - left_wrist_roll_joint
        - left_gripper_joint
    baselink: left_base_link
    endlink: left_end_roll_link
    differential_pairs: [1, 2, 3, 4, 5, 6]
    gear_ratios: [7.1875, 7.1875, 8.2444852941, 7.1875, 8.24448529412, 7.1875, 8.24448529412, 14.4]
    joint_torque_directions: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
    id_torque_gains: [1.12, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
    # id_torque_gains: [1.38, 1.27, 1.27, 1.27, 1.27, 1.0, 1.0]
    softstop_min_angles: [-2.2, -2.25, -2.6, -2.25, -2.6, -2.25, -2.6, -1.45]
    softstop_max_angles: [3.3, 0.5, 2.6, .5, 2.6, 0.5, 2.6, 1.45]
    softstop_tolerance: 0.19
    softstop_torque_limit: 95.0

    ### Motor configuration
    #serial_port: /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A506NO9F-if00-port0
    serial_port: /dev/serial/by-id/usb-FTDI_FT232R_USB_UART_AI057K87-if00-port0
    motor_names:
        - base_roll_motor
        - right_motor1
        - left_motor1
        - right_motor2
        - left_motor2
        - right_motor3
        - left_motor3
        - gripper_motor
    motor_ids: [14, 43, 38, 49, 46, 41, 42, 2]
    # motor_current_limits: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    motor_current_limits: [2.0, 2.0, 2.0, 2.0, 2.0, 1.0, 1.0, 0.3]
    current_to_torque_ratios: [0.626, 0.626, 0.626, 0.626, 0.626, 1.0, 1.0, 1.0]

    ### Calibration configuration
    hardstop_start_angles: [-0.785398, -2.19, -1.570796, 0.0, 1.570796, -0.23, 0.0, 0.0]
    calibration_initial_angles: [-0.62, 0.99, 0.0, -0.83, 0.0, 0.0, 0.0, 0.0] # for vive calibration

    ### Posture control parameters for gradient descent ik pose controller
    posture_target: [0.9, -1.57, 0.0, -1.57, 0.90, -1.0, 0.0]
    posture_weights: [0.2, 1, 1, 1, 1, 0.2, 0.5]
